packages/internal-workflow-common/src/errors.ts

```diff
@@ -6,12 +6,8 @@ export class PayloadConverterError extends ValueError {
   public readonly name: string = 'PayloadConverterError';
 }
 
-export class UnsupportedJsonTypeError extends PayloadConverterError {
-  public readonly name: string = 'UnsupportedJsonTypeError';
-
-  constructor(message: string | undefined, public readonly cause?: Error) {
-    super(message ?? undefined);
-  }
+export class UnsupportedTypeError extends PayloadConverterError {
```

Comment:
Why not reuse TypeError?

Reply:
Generally, I'd make more specific errors in case a dev would want to do `instanceof` to differentiate. In this case, I don't imagine that being common—would be useful if the dev doesn't know what type of arg they're sending (eg it's provided by the user or something).